pkg/controllers/management/template/template_cluster_catalog.go

```diff
@@ -0,0 +1,190 @@
+package template
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+var rolesWithAccessToClusterCatalogTemplates = map[string]bool{
+	"cluster-owner":          true,
+	"cluster-member":         true,
+	"clustercatalogs-manage": true,
+	"clustercatalogs-view":   true,
+}
+
+func (tm *RBACTemplateManager) reconcileClusterCatalog(clusterID string) error {
+	var clusterTemplates, clusterTemplateVersions []string
+	// Get all cluster catalogs created in this cluster
+	clusterCatalogs, err := tm.clusterCatalogLister.List(clusterID, labels.NewSelector())
+	if err != nil {
+		log.Errorf("Error while listing cluster catalogs: %v", err)
```

Comment:
no need to log in the method. Just return an error to the framework - the error can be customized of course, just not logged